Public/Src/Tools/Execution.Analyzer/Analyzers.Partner/PipExecutionPerformanceAnalyzer.cs

```diff
@@ -56,23 +63,67 @@ internal sealed class PipExecutionPerformanceAnalyzer : Analyzer
     {
         private const string Indentation = "    ";
         private readonly StreamWriter m_writer;
+        private readonly bool m_isCsvFormat;
+
         private readonly Dictionary<PipId, ProcessPipExecutionPerformance> m_processPerformance =
             new Dictionary<PipId, ProcessPipExecutionPerformance>();
 
+        private Dictionary<PipId, TimeSpan> m_stepDurations = new Dictionary<PipId, TimeSpan>();
+
         private string m_indent = string.Empty;
 
         /// <nodoc />
         [SuppressMessage("Microsoft.Reliability", "CA2000:Dispose objects before losing scope")]
-        public PipExecutionPerformanceAnalyzer(AnalysisInput input, string outputFilePath)
+        public PipExecutionPerformanceAnalyzer(AnalysisInput input, string outputFilePath, bool isCsvFormat)
             : base(input)
         {
             Contract.Requires(!string.IsNullOrWhiteSpace(outputFilePath));
 
             m_writer = new StreamWriter(outputFilePath);
+            m_isCsvFormat = isCsvFormat;
         }
 
-        /// <inheritdoc />
         public override int Analyze()
+        {
+            if (m_isCsvFormat)
+            {
+                ToCsv();
+            }
+            else
+            {
+                ToJson();
+            }
+
+            return 0;
+        }
+
+        private void ToCsv()
+        {
+            m_writer.WriteLine("Description,Step,Wall,User,Kernel,User/Wall,ReadByes,WriteBytes,OtherBytes,Processes,Memory");
+            foreach (var processIdAndExecutionPerformance in m_processPerformance.OrderByDescending(pip => pip.Value.ProcessExecutionTime.TotalMilliseconds))
+            {
+                var process = CachedGraph.PipGraph.GetPipFromPipId(processIdAndExecutionPerformance.Key) as Process;
+                Contract.Assert(process != null);
+
+                var performance = processIdAndExecutionPerformance.Value;
+
+                m_writer.Write(process.GetShortDescription(CachedGraph.Context) + ",");

```

Comment:
>"," [](start = 82, length = 3)

Since you are using individual writes, you could avoid the string concatenation and the allocations for ath you could write this as `m_writer.Write(expr); m_writer.Write(',');` (or perhaps with a helper method `WriteColum(string expr, bool isLast = false)`